### PR TITLE
Enhancement/guided tour/disable tour for mobile

### PR DESCRIPTION
**Pull request by lovaiible**
### Checklist
- [x] I tested the changes and all related features on the test server https://artemistest.ase.in.tum.de.
- [x] ~~I documented my source code using the JavaDoc / JSDoc style.~~
- [x] ~~I added integration test cases for the server (Spring) related to the features~~
- [x] ~~I added integration test cases for the client (Jest) related to the features~~
- [x] I added screenshots/screencast of my UI changes
- [x] ~~I translated all the newly inserted strings~~

### Description
The guided tour is currently not optimized for mobile devices. Therefore I disabled the `Start Tour` button for mobile phones and tablets and added the `//TODO` remark for a future task.

### Steps for Testing
1. Log in to Artemis
2. Click on account-menu
3. Start tour from account menu
4. Redo the steps from a mobile phone or tablet
5. The `Start tour` menu item should not be visible

### Screenshots
IMG_9856

### CODE REVIEW

File: src/main/webapp/app/layouts/navbar/navbar.component.ts [hunk 1]
```diff
@@ -80,7 +80,13 @@ export class NavbarComponent implements OnInit, OnDestroy {
     subscribeForGuidedTourAvailability(): void {
         // Check availability after first subscribe call since the router event been triggered already
         this.guidedTourService.getGuidedTourAvailabilityStream().subscribe(isAvailable => {
-            this.isTourAvailable = isAvailable;
+            // The guided tour is currently disabled for mobile devices and tablets
+            // TODO optimize guided tour layout for mobile devices and tablets
+            if (this.deviceService.isMobile() || this.deviceService.isTablet()) {
```
Comment (jpbernius) [inline src/main/webapp/app/layouts/navbar/navbar.component.ts]: For me, it feels like this is the wrong place to add this check.
I would suggest to move this to a `.map` in the [guided-tour.service.ts#L82].

[guided-tour.service.ts#L82]: https://github.com/ls1intum/Artemis/blob/develop/src/main/webapp/app/guided-tour/guided-tour.service.ts#L82

Comment (lovaiible) [reply src/main/webapp/app/layouts/navbar/navbar.component.ts]: Thank you for the suggestion, I moved the logic to the `GuidedTourService` now.

File: src/main/webapp/app/guided-tour/guided-tour.service.ts [hunk 2]
```diff
@@ -79,7 +86,9 @@ export class GuidedTourService {
      * @return Observable(true) if the guided tour is available for the current component, otherwise Observable(false)
      */
     public getGuidedTourAvailabilityStream(): Observable<boolean> {
-        return this.guidedTourAvailability.asObservable();
+        // The guided tour is currently disabled for mobile devices and tablets
+        // TODO optimize guided tour layout for mobile devices and tablets
```
Comment (thilo-behnke) [inline src/main/webapp/app/guided-tour/guided-tour.service.ts]:
```suggestion
        // TODO: optimize guided tour layout for mobile devices and tablets
```

Comment (jpbernius) [reply src/main/webapp/app/guided-tour/guided-tour.service.ts]: Sorry, missed that one :D

**Pull request merged**